sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiver.cs

```diff
@@ -194,15 +223,22 @@ public class ServiceBusReceiver : IAsyncDisposable
         {
             Argument.AssertNotClosed(IsClosed, nameof(ServiceBusReceiver));
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            ServiceBusEventSource.Log.ReceiveMessageStart(Identifier, maxMessages);
+            IList<ServiceBusReceivedMessage> messages = null;
 
             try
             {
-                return await _innerReceiver.ReceiveBatchAsync(maxMessages, cancellationToken).ConfigureAwait(false);
+                messages = await _innerReceiver.ReceiveBatchAsync(maxMessages, cancellationToken).ConfigureAwait(false);
             }
-            finally
+            catch (Exception exception)
             {
-                // TODO: Add log - SeviceBusEventSource.Log.ReceiveBatchAsyncComplete();
+                ServiceBusEventSource.Log.ReceiveMessageException(Identifier, exception);
+                throw;
             }
+
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            ServiceBusEventSource.Log.ReceiveMessageComplete(Identifier, maxMessages);
```

Comment:
I was intentionally leaving these out of the finally block since if an exception is thrown we probably don't want to log that the operation was completed.

Reply:
Same as above.